### ISSUE TITLE
Double clicking a symbol starting/ending a string highlights the adjacent quotes

### ISSUE THREAD

**Issue by willfaught** [Microsoft/vscode#22685]
When writing JavaScript, double clicking symbols that start a string highlights both the symbols and the single or double quote to the left of it. If there's a left parenthesis to the left of the quote, that is highlighted too. The reverse happens for symbols ending a string, except the highlight also extends to a line-terminating semicolon if it's to the right of the right parenthesis.

If the first or last word are only letters, the selection is only of the words themselves, as expected.

- VSCode Version: Version 1.10.2 (1.10.2), 8076a19fdcab7e1fc1707952d652f0bb6c6db331
- OS Version: MacOS 10.12.3 (16D32)

Steps to Reproduce:

1. Enter line: `console.log('### ###');`
2. Double click either the first `###` or the second to observe.
3. Also try `console.log('######');` and `console.log('###abc###');`

Probably related: If the cursor is positioned in one of those leading or trailing string symbols and you use Option-Left/Right to skip to the end of that word at the string boundary, it will also skip over the string quote, parenthesis, and semicolon, unlike if it was letters only.

**Comment by alexandrudima** [comment]
This is as designed.

Double clicking and word navigation respects the user configurable setting:
```
  "editor.wordSeparators": "`~!@#$%^&*()-=+[{]}\\|;:'\",.<>/?",
```

These are programming language agnostic features that simply use the word separators list. i.e. double clicking does not analyze the underlying language and determine it to be JavaScript, and implement a different strategy than under C++ for example, where `'` can only contain a single character, while in JavaScript `'` can contain more than one character. 

In other words, this operation on this text works the same, regardless of file extension. Would you like for double clicking and word navigation to be language dependentant?

**Comment by willfaught** [comment]
If apostrophe is already in wordSeparators, then why does the selection include it?

Either way, I think it should be smart about editing code versus English or some human language. Whether that involves per-language word separator configuration, I'm sure you know better than me! :)

**Comment by alexandrudima** [comment]
In this case, the explanation is easy, `#` is also a word separator. 

Double clicking works by classifying characters in one of three classes: regular (highlighted in green below), word separators (highlighted in blue below) and whitespace (highlighted in red below).

![image](https://cloud.githubusercontent.com/assets/5047891/24492098/779c1b6c-152a-11e7-85f0-063479c69cad.png)


Then, it will select contiguous characters of the same class in both directions starting from the double click location. 

That's why double clicking on `#` will select adjacent word separator characters.

I have also double checked in Visual Studio and I believe what we do is correct and consistent with other editors. I will close this for now, since it works as designed.

**Comment by willfaught** [comment]
Makes sense. Thanks for the explanation (and graphic!).

Perhaps something to consider changing in the future?
